src/Views/VPNPage.vala

```diff
@@ -80,29 +62,53 @@ public class Network.VPNPage : Network.Widgets.Page {
         add_button.tooltip_text = _("Add VPN Connection…");
         add_button.clicked.connect (() => {
             add_button.sensitive = false;
+            add_button.sensitive = create_connection ();
+        });
 
-            try {
-                var command = AppInfo.create_from_commandline ("nm-connection-editor --create --type=vpn", null, GLib.AppInfoCreateFlags.NONE);
-                command.launch (null, null);
-
-                add_button.sensitive = true;
-            } catch (Error e) {
+        var remove_button = new Gtk.ToolButton (new Gtk.Image.from_icon_name ("list-remove-symbolic", Gtk.IconSize.SMALL_TOOLBAR), null);
+        remove_button.tooltip_text = _("Forget selected VPN…");
+        remove_button.sensitive = false;
+        //  remove_button.clicked.connect (delete_connection);
+        remove_button.clicked.connect (() => {
+            sel_row = vpn_list.get_selected_row () as VPNMenuItem;
```

Comment:
You should verify if the selected row isn't null here to not access null properties.